### DISCUSSION TITLE
Why is the event for cursor being over a control called PointerEnter but called pointerover when selecting it with a style?

### DISCUSSION THREAD

**Discussion by YoriMirus** [AvaloniaUI/Avalonia#7690] [Q&A] [answered]
Hello. I'm new to Avalonia and I was trying to add an animation to a button so that when the cursor is over it, the background changes from a specific background to Red.

I know that the name for the event is PointerEnter so I named the selector "Button.CloseButton:PointerEnter", however, that did not work. After looking on the internet, I found out you are supposed to use "pointerover" instead of "PointerEnter". Also it seems like you can't use capital letters, so "PointerOver" doesn't work either.

I would like to know. Why is it this way? How do I know which event corresponds to which name when selecting it?

**Comment by jp2masa** [comment, marked as answer]
Pseudoclasses correspond to states, not events. So in this case, they're separate things: the `PointerEnter` event is raised when the pointer enters the control, and there's a particular handler which enables the `pointerover` state, which can be used in selectors.

**Reply by YoriMirus** [reply]
I understand, is there a way to "select" an event if there is no pseudoclass for it? Or is there a pseudoclass for each event?

**Reply by jp2masa** [reply]
I don't think so, you would have to implement a handler for that. For `pointerover` it's implemented like this:

1. [`IsPointerOver` property](https://github.com/AvaloniaUI/Avalonia/blob/42852e50b1dd8db7304d7f76e1dbcd7d14946ed0/src/Avalonia.Input/InputElement.cs#L68-L72)
2. [`PointerEnter`/`PointerLeave` which set `IsPointerOver` value](https://github.com/AvaloniaUI/Avalonia/blob/42852e50b1dd8db7304d7f76e1dbcd7d14946ed0/src/Avalonia.Input/InputElement.cs#L653-L671)
3. [`OnPropertyChanged` handler](https://github.com/AvaloniaUI/Avalonia/blob/42852e50b1dd8db7304d7f76e1dbcd7d14946ed0/src/Avalonia.Input/InputElement.cs#L629-L632)
4. [`UpdatePseudoClasses`](https://github.com/AvaloniaUI/Avalonia/blob/42852e50b1dd8db7304d7f76e1dbcd7d14946ed0/src/Avalonia.Input/InputElement.cs#L704-L707), which is called on the [constructor as well to set initial state](https://github.com/AvaloniaUI/Avalonia/blob/42852e50b1dd8db7304d7f76e1dbcd7d14946ed0/src/Avalonia.Input/InputElement.cs#L228-L231)

For item 2. there may be a virtual method which you can override to handle your event, but if there isn't or you don't want to use it you can always register a handler [like this](https://github.com/AvaloniaUI/Avalonia/blob/42852e50b1dd8db7304d7f76e1dbcd7d14946ed0/src/Avalonia.Input/InputElement.cs#L219-L220).

**Reply by YoriMirus** [reply]
I see thank you very much.
